### PR TITLE
fix(cli): Change the project root semantics such as `grafbase/` becomes only a fallback for the project root

### PR DESCRIPTION
**Pull request by jakubadamw**
# Description

Introduce a fallback mechanism in searching the paths such as the current `grafbase` subdirectory remains supported. It will be phased out in the future. At the same time, the main search path for 'schema.graphql' and 'grafbase.config.ts' will now be the current directory.

# Type of change

- [ ] 💔 Breaking
- [X] 🚀 Feature
- [ ] 🐛 Fix
- [ ] 🛠️ Tooling
- [ ] 🔨 Refactoring
- [ ] 🧪 Test
- [ ] 📦 Dependency
- [ ] 📖 Requires documentation update

### CODE REVIEW

File: cli/crates/common/src/environment.rs [hunk 1]
```diff
@@ -195,20 +192,15 @@ impl Project {
...
 
-        let grafbase_directory_path = schema_path
-            .parent()
-            .expect("the schema directory must have a parent by definiton")
-            .to_path_buf();
-
-        let path = grafbase_directory_path
+        let path = schema_path
             .parent()
             .expect("the grafbase directory must have a parent directory by definition")
```
Comment (tomhoule) [inline cli/crates/common/src/environment.rs:197]: Do we always expect a parent here? It seems unlikely that users would have their project at the filesystem root, but it's not unthinkable.

**Review by tomhoule — APPROVED** [review]
I'm not the most familiar with the CLI internals but this looks good :+1:

File: cli/crates/common/src/environment.rs (hunk 1 above)
Comment (jakubadamw) [reply cli/crates/common/src/environment.rs:197]: `schema_path` would then be `/schema.graphql` or `/grafbase.config.ts`, so `parent()` for those is '/'.

Comment (tomhoule) [reply cli/crates/common/src/environment.rs:197]: Ah yes, good point, all good then :+1:

**Pull request merged**